src/controllers/update/update.filing.date.controller.ts

```diff
@@ -56,3 +57,20 @@ export const post = async(req: Request, res: Response, next: NextFunction) => {
     next(error);
   }
 };
+
+const getFilingDate = async (req: Request, appData: ApplicationData): Promise<{} | undefined> => {
+  // use date stored in appData if present
+  let filingDate = appData.update?.[FilingDateKey] ? mapDataObjectToFields(appData.update[FilingDateKey], FilingDateKeys, InputDateKeys) : undefined;
+
+  // otherwise use the next made up to date from confirmation statement in company profile
+  if (!filingDate && appData.entity_number) {
```

Comment:
entity_number is of type 'string | undefined' but getConfirmationStatementNextMadeUpToDateAsIsoString requires a string, so the null check was to keep the compiler happy so it knows it will be a string passed into the function and not undefined. I could add a specific null check and throw a new error instead.

Reply:
I think doing the specific check and throwing an error would be better. At the moment it looks like we're testing for `undefined` but not actually dealing with it when it is `undefined`.

Reply:
added the extra check